mobs/nakki_boss/LilypadAttackStats.cs

```diff
@@ -0,0 +1,94 @@
+using Godot;
+using Godot.Collections;
+
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Jakojaannos.WisperingWoods;
+
+public partial class LilypadAttackStats(LilypadSelectionStrategy selectionStrategy) : Resource {
+	public float UnderwaterTime { get; set; } = 1.5f;
+	public float UnderwaterTimeVariation { get; set; } = 0.5f;
+	public float SinkAnimationSpeed { get; set; } = 1.0f;
+	public float SinkAnimationSpeedVariation { get; set; } = 0.5f;
+	public float ShakeTime { get; set; } = 0.75f;
+	public float ShakeTimeVariation { get; set; } = 0.5f;
+	public int AttackId { get; set; } = GenerateId();
```

Comment:
Having a generated ID turns the `Stats` into an `Instance`. I'd rather keep the running IDs implementation detail of Something Somewhere, instead of relying on them all over the place.

That is, the reliance one the IDs should be contained to as narrow scope as possible (e.g. an implementation detail of a single class, if possible) and there should be e.g. a `struct AttackInstance { Stats stats; int AttackId = GenerateId(); }` wrapper.

The "public-facing" stats struct/resource should not contain implementation details of the underlying attack management system.

Reply:
Looking at this again, these never seem to be exposed publicly, so these actually are more of "attack instances" than "attack stats". In case these are going to be publicly exposed and configurable, I guess these should be split to separate classes for "stats" (with base values, variations etc.) and the actual attack instance (with just the final randomized values used for executing the attack).